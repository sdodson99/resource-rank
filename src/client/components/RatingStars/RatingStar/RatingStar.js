import React from 'react';
import PropTypes from 'prop-types';
import styles from './RatingStar.module.css';
import HalfRatingStar from '../HalfRatingStar/HalfRatingStar';
import FullRatingStar from '../FullRatingStar/FullRatingStar';
import EmptyRatingStar from '../EmptyRatingStar/EmptyRatingStar';

const RatingStar = ({ fill, size, color }) => {
  const getRatingStarForFill = () => {
    const isHalfStar = fill >= 0.3 && fill <= 0.7;
    if (isHalfStar) {
      return <HalfRatingStar size={size} color={color} />;
    }

    const isFullStar = fill > 0.7;
    if (isFullStar) {
      return <FullRatingStar size={size} color={color} />;
    }

    return <EmptyRatingStar size={size} />;
  };

  return (
    <div className={styles.RatingStar} data-testid="RatingStar">
      {getRatingStarForFill()}
    </div>
  );
};

RatingStar.propTypes = {
  fill: PropTypes.number,
  size: PropTypes.number,
  color: PropTypes.string,
};

RatingStar.defaultProps = {
  fill: 0,
  size: 20,
  color: 'black',
};

export default RatingStar;
