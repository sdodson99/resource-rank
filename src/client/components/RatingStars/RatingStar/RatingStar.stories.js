import React from 'react';
import RatingStar from './RatingStar';

export default {
  component: RatingStar,
  title: 'Components/RatingStars/RatingStar',
  args: {
    size: 50,
  },
};

const Template = (args) => <RatingStar {...args} />;

export const Default = Template.bind({});
Default.args = {};

export const Color = Template.bind({});
Color.args = {
  color: 'orange',
  fill: 1,
};

export const Size = Template.bind({});
Size.args = {
  ...Color.args,
  size: 100,
  fill: 1,
};

export const Empty = Template.bind({});
Empty.args = {
  ...Color.args,
  fill: 0,
};

export const Half = Template.bind({});
Half.args = {
  ...Color.args,
  fill: 0.5,
};

export const Full = Template.bind({});
Full.args = {
  ...Color.args,
  fill: 1,
};
