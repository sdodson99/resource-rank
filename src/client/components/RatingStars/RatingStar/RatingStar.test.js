import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import RatingStar from './RatingStar';
import { createRenderer } from 'react-test-renderer/shallow';

describe('<RatingStar />', () => {
  it('should mount', () => {
    render(<RatingStar />);

    const ratingStar = screen.getByTestId('RatingStar');

    expect(ratingStar).toBeInTheDocument();
  });

  it('should render correctly', () => {
    const component = createRenderer().render(
      <RatingStar fill={0.5} color={'orange'} size={50} />
    );

    expect(component).toMatchSnapshot();
  });

  it('should render correctly for empty rating', () => {
    const component = createRenderer().render(<RatingStar fill={0} />);

    expect(component).toMatchSnapshot();
  });

  it('should render correctly for half rating', () => {
    const component = createRenderer().render(<RatingStar fill={0.5} />);

    expect(component).toMatchSnapshot();
  });

  it('should render correctly for full rating', () => {
    const component = createRenderer().render(<RatingStar fill={1} />);

    expect(component).toMatchSnapshot();
  });
});
