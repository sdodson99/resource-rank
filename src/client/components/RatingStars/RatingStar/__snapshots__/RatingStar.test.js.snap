// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RatingStar /> should render correctly 1`] = `
<div
  className="RatingStar"
  data-testid="RatingStar"
>
  <HalfRatingStar
    color="orange"
    size={50}
  />
</div>
`;

exports[`<RatingStar /> should render correctly for empty rating 1`] = `
<div
  className="RatingStar"
  data-testid="RatingStar"
>
  <EmptyRatingStar
    size={20}
  />
</div>
`;

exports[`<RatingStar /> should render correctly for full rating 1`] = `
<div
  className="RatingStar"
  data-testid="RatingStar"
>
  <FullRatingStar
    color="black"
    size={20}
  />
</div>
`;

exports[`<RatingStar /> should render correctly for half rating 1`] = `
<div
  className="RatingStar"
  data-testid="RatingStar"
>
  <HalfRatingStar
    color="black"
    size={20}
  />
</div>
`;
